### DISCUSSION TITLE
Graphql Subscription

### DISCUSSION THREAD

**Discussion by shiftlabs1** [exogee-technology/graphweaver#176] [General]
Hi I stumbled on this project yesterday and have since been playing with it using an existing postgres database and i have been impressed so far. I wanted to ask if there is any plans for graphql subscriptions and if yes , what is the planned approach (SSE or WS) . Also How soon is it planned for. I will very much start on a new project with this. 
Thanks

**Comment by thekevinbrown** [comment]
Hey there, thanks for the kind words, hope it's as useful for you as it has been for us!

It's definitely on the roadmap. We haven't yet had a strong reason in our projects to implement them yet, but we will likely need them for our own uses soon.

Out of curiosity, can you tell me more about your use case for subscriptions in your application? In particular, what backend(s) would the data live in that you want to subscribe to?

Given that we're focused on Serverless tech for this, we'd end up building them with [API Gateway's Web Sockets support](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-api-overview.html), and change notifications would need to be supported by the backend storage platform. If it's a database that we're managing, then that'll be supported with subscriptions out of the box, but if it's a REST API, OData, or similar, it'd need to have a way of notifying us of changes to the underlying data as well.

**Reply by shiftlabs1** [reply]
Sorry for the late reply. My backend storage is mainly postgres at the moment. I do not expect subscriptions implementation for REST /OData unless as an aggregate response to some DB changes which will have to be custom implementation . Our use case is one of providing real time data to some demographic/section of app users (specifically distributed business hubs) , while standard queries and mutations work for all other use cases we need the subscriptions to provide realtime feedback without constant refreshes (queries)

**Reply by thekevinbrown** [reply]
Yup, cool, that makes a lot of sense, thanks! That aligns with how we would use subscriptions if we had them as well, e.g. Postgres data -> Subscription -> Frontend
